src/main/java/com/google/sps/servlets/AddOrganizationServlet.java

```diff
@@ -17,65 +17,77 @@
 import java.util.ArrayList;
+import java.text.SimpleDateFormat;
 import com.google.sps.data.User;
+import com.google.sps.data.HistoryManager;
+import com.google.sps.data.OrganizationUpdater;
+import com.google.sps.data.GivrUser;
+import java.time.Instant;
 import java.io.IOException;
 
 @WebServlet("/add-organization")
 public class AddOrganizationServlet extends HttpServlet {
 
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // for now, ID will just be the one datastore gives it automatically
-    String orgName = request.getParameter("org-name");
-    String orgEmail = request.getParameter("email");
-    String orgStreetAddress = request.getParameter("address");
-    String orgPhoneNum = request.getParameter("phone-number");
-    String orgUrl = request.getParameter("url-link");
-    String orgDescription = request.getParameter("description");
-    String hourOpen = request.getParameter("hour-open");
-    String hourClosed = request.getParameter("hour-closed");
+    GivrUser user = GivrUser.getLoggedInUser();
 
-    //TODO: get timestamp with transactions instead
-    long timestampMillis = System.currentTimeMillis();
+    if (user.getUserId().equals("")) {
+      throw new IllegalArgumentException("Error: unable to register organization if user is not logged in.");
+    }
+    
+
+    /* MillisecondSinceEpoch represent the number of milliseconds that have passed since
+     * 00:00:00 UTC on January 1, 1970. It ensures that all users are entering a representation
+     * of time that is independent of their time zone */
+    long millisecondSinceEpoch = Instant.now().toEpochMilli();
 
     // when suppliers are added, Entity kind will be from a parameter- for now is hardcoded
-    Entity newOrganization = new Entity("Distributor");
+    Entity newOrganizationEntity = new Entity("Distributor");
 
-    // for now just taking open & closed hours from form, not checking for null or that closed time > open
+    /* This implementation stores history entries as embedded entities instead of custom objects
+     * because it is much simpler that way */
+    ArrayList changeHistory = new ArrayList<>();
+    
+    HistoryManager history = new HistoryManager();
 
-    // This implementation has 0 = 12:00AM, 1 = 1:00AM, 13 = 1:00PM, etc. 
-    ArrayList<Integer> openHours = new ArrayList<Integer>();
-    openHours.add(Integer.parseInt(hourOpen));
-    openHours.add(Integer.parseInt(hourClosed));
+    changeHistory.add(history.recordHistory("Organization was registered", millisecondSinceEpoch));
+    newOrganizationEntity.setProperty("changeHistory", changeHistory);
 
+    //TODO use UserId's here
     ArrayList<String> moderatorList = new ArrayList<String>();
-    moderatorList.add("anon creator"); //dummy value for before auth gets set up
+    moderatorList.add(user.getUserId());
+
+    newOrganizationEntity.setProperty("creationTimeStampMillis", millisecondSinceEpoch);
```

Comment:
Leave a comment as to why you are setting these directly here instead of organization updater?